hathor/indexes/deps_index.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2021 Hathor Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, Iterator, List
+
+from hathor.transaction import BaseTransaction, Block
+
+if TYPE_CHECKING:  # pragma: no cover
+    from hathor.transaction.storage import TransactionStorage
+
+
+# XXX: this arbitrary height limit must fit in a u32 (4-bytes unsigned), so it can be stored easily on rocksdb
+INF_HEIGHT: int = 2**32 - 1
+
+
+def get_requested_from_height(tx: BaseTransaction) -> int:
+    """Return the height of the block that requested (directly or indirectly) the download of this transaction.
+
+    If this value cannot be determined (either for the lack of a metadata or otherwise), the INF_HEIGHT constant is
+    returned instead. So there will always be a value (it's never None).
+
+    This is used to help prioritize which transaction to start from next on sync-v2 when syncing the transactions
+    after downloading a chain of blocks.
+    """
+    assert tx.storage is not None
+    if tx.is_block:
+        return tx.get_metadata().height
+    first_block = tx.get_metadata().first_block
+    if first_block is None:
+        # XXX: consensus did not run yet to update first_block, what should we do?
+        #      I'm defaulting the height to `inf` (practically), this should make it heightest priority when
+        #      choosing which transactions to fetch next
+        return INF_HEIGHT
+    else:
```

Comment:
You don't need this `else`, right?

Reply:
Right. I'll remove it, but I don't think it matters too much. I'd prefer if we used `black` and have it pick a style to use uniformly. I think I'm more used to do `if: ... return; else: ... return` than `if: ... return; ... return`, but it's just a matter of style anyway.

Reply:
I agree. I prefer the second way because I feel the if statement is a guard clause.

Reply:
Done.

Reply:
Thanks!